great_expectations/profile/basic_dataset_profiler.py

```diff
@@ -0,0 +1,117 @@
+from .base import DataSetProfiler
+
+
+class BasicDatasetProfiler(DataSetProfiler):
+    """A profiler inspired by the beloved pandas_profiling project
+    """
+
+    @classmethod
+    def _get_column_type_and_cardinality(cls, df, column):
+
+        if df.expect_column_values_to_be_of_type(column, "int")["success"]:
+            type_ = "int"
+
+        elif df.expect_column_values_to_be_of_type(column, "float")["success"]:
+            type_ = "float"
+
+        elif df.expect_column_values_to_be_of_type(column, "string")["success"]:
+            type_ = "string"
+
+        else:
+            type_ = "unknown"
+
+        unique = df.expect_column_unique_value_count_to_be_between(column, 0, None)[
```

Comment:
This is an expensive computation that is never used...shouldn't it be dropped?

Reply:
Let's leave that for later. Fleshing out specifics of this profiler aren't the main goal for this PR.

Reply:
Clarification on this point --> Agreed that we don't need to fully flesh out the profiler, but I think we should not merge in totally dead/unreachable code...if there's a different desired behavior it's easier to add later than to remember to go back and pull out chunks

Reply:
as discussed, commenting out.